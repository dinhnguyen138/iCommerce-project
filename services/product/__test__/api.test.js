'use strict';
require('mockingoose').default;
const axios = require('axios');
const request = require('supertest');
const server = require('../server.js');
jest.mock('axios');

axios.get.mockImplementation( input => {
  return;
});

beforeAll(async () => {
  console.log('Jest starting!');
});
beforeEach(async () => {
});
afterAll(() => {
  server.close();
  console.log('server closed!');
});

describe('API test', () => {
  test('GET /ping test', async () => {
    const response = await request(server).get('/ping');
    expect(response.status).toEqual(200);
    expect(response.text).toMatchSnapshot();
  });
  test('GET / default test', async () => {
    const response = await request(server).get('/');
    expect(response.status).toEqual(200);
    expect(response.text).toMatchSnapshot();
  });
  test('GET / with full text seach test', async () => {
    const response = await request(server).get('/?query=cannon');
    expect(response.status).toEqual(200);
    expect(response.text).toMatchSnapshot();
  });
  test('GET /find/:id test', async () => {
    const response = await request(server).get('/find/1');
    expect(response.status).toEqual(404);
    expect(response.text).toMatchSnapshot();
  });
});