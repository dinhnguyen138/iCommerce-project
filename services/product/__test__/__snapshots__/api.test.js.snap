// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API test GET / default test 1`] = `"{}"`;

exports[`API test GET / with full text seach test 1`] = `"{}"`;

exports[`API test GET /find/:id test 1`] = `"Product not found"`;

exports[`API test GET /ping test 1`] = `"Ping success"`;
